test/controllers/PenaltyDetailsController.test.ts

```diff
@@ -0,0 +1,48 @@
+import 'reflect-metadata'
+import { createApplication } from '../ApplicationFactory';
+import * as request from 'supertest'
+import { createSubstituteOf } from '../SubstituteFactory';
+
+import '../../src/controllers/PenaltyDetailsController';
+import { RedisService } from '../../src/services/RedisService';
+import { BAD_REQUEST, OK } from 'http-status-codes';
+import { PenaltyReferenceDetails } from '../../src/models/PenaltyReferenceDetails';
+
+const app = createApplication(container => {
+    container
+        .bind(RedisService).toConstantValue(createSubstituteOf<RedisService>(service => {
+            service.ping().returns(true)
+        }));
+});
+
+describe('PenaltyDetailsController', () => {
+
+    it('should return 200 when trying to access the penalty-reference page', async () => {
+        await request(app)
+            .get('/penalty-reference')
+            .expect(OK);
+    });
+
+    // it('should return 200 when posting valid penalty details', async () => {
+    //     const penaltyDetails: PenaltyReferenceDetails = {
+    //         penaltyReference: 'A12345678',
+    //         companyNumber: 'SC123123'
+    //     };
+    //     await request(app)
+    //         .post('/penalty-reference')
+    //         .send(penaltyDetails)
+    //         .expect(OK);
+    // });
+
+    it('should return 400 when posting empty penalty reference', async () => {
+        const penaltyDetails: PenaltyReferenceDetails = {
+            penaltyReference: '',
+            companyNumber: 'SC123123'
+        };
+        await request(app)
+            .post('/penalty-reference')
+            .send(penaltyDetails)
+            .expect(BAD_REQUEST);
```

Comment:
Please change it to 422 not 400. A 400 is for `The server could not understand the request due to invalid syntax.` while 422 is for `The request was well-formed but was unable to be followed due to logic errors.`. Request with invalid form data are valid from the syntax standpoint, they are just not acceptable from an application point of view.

Other definitions:

400 - The request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications.

422 - The 422 (Unprocessable Entity) status code means the server understands the content type of the request entity (hence a 415(Unsupported Media Type) status code is inappropriate), and the syntax of the request entity is correct (thus a 400 (Bad Request) status code is inappropriate) but was unable to process the contained instructions. For example, this error condition may occur if an XML request body contains well-formed (i.e., syntactically correct), but semantically erroneous, XML instructions.

Reply:
👍 - thanks for the explanation,  i've updated this